ext/posix/posix.c

```diff
@@ -516,8 +516,9 @@ PHP_FUNCTION(posix_isatty)
 		if (!zend_parse_arg_long(z_fd, &fd, /* is_null */ NULL, /* check_null */ false, /* arg_num */ 1)) {
 			php_error_docref(NULL, E_WARNING, "Argument #1 ($file_descriptor) must be of type int|resource, %s given",
 				zend_zval_value_name(z_fd));
-			fd = zval_get_long(z_fd);
+			RETURN_FALSE;
 		}
+		fd = zval_get_long(z_fd);
```

Comment:
Actually, this part looks redundant, no? `fd` is written to by `zend_parse_arg_long`, but only if successful.

Reply:
Ah indeed!